client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -311,25 +320,34 @@ class DeltaSharingRestClient(
     val target = getTargetUrl(
       s"/shares/$encodedShareName/schemas/$encodedSchemaName/tables/$encodedTableName/metadata" +
         s"$encodedParams")
-    val (version, respondedFormat, lines) = getNDJson(target)
+    val response = getNDJson(target)
 
     checkRespondedFormat(
-      respondedFormat,
+      response.respondedFormat,
       rpc = "getMetadata",
       table = s"${table.share}.${table.schema}.${table.name}"
     )
 
-    if (respondedFormat == RESPONSE_FORMAT_DELTA) {
-      return DeltaTableMetadata(version, lines = lines, respondedFormat = respondedFormat)
+    if (response.respondedFormat == RESPONSE_FORMAT_DELTA) {
+      return DeltaTableMetadata(
+        response.version,
+        lines = response.lines,
+        respondedFormat = response.respondedFormat
+      )
     }
 
-    val protocol = JsonUtils.fromJson[SingleAction](lines(0)).protocol
+    val protocol = JsonUtils.fromJson[SingleAction](response.lines(0)).protocol
     checkProtocol(protocol)
-    val metadata = JsonUtils.fromJson[SingleAction](lines(1)).metaData
-    if (lines.size != 2) {
+    val metadata = JsonUtils.fromJson[SingleAction](response.lines(1)).metaData
+    if (response.lines.size != 2) {
       throw new IllegalStateException("received more than two lines")
     }
```

Comment:
Shall we move this check to after `checkRespondedFormat`? It should be performed for both parquet and delta format.

Reply:
moved, but seems an independent fix.